test/clj_kondo/impl/parser_test.clj

```diff
@@ -22,8 +22,31 @@
                 (and (Double/isInfinite thing)
                      (< thing 0))))))
 
+(defn- parse-error
+  "Parse the source, which is expected to contain a syntax error, and return the
+  error message produced from rewrite-clj."
+  [source]
+  (try
+    (parse-string source)
+    nil
+    (catch Exception e
+      (.getMessage e))))
+
+(deftest parse-string-test
+  ;; This test has every syntax error that can cause rewrite-clj to throw using
+  ;; the `clj-kondo.impl.rewrite-clj.parser.utils/throw-reader` function.
+  ;; This allows us to test for regressions when that function is refactored.
+  (are [source message] (= message (parse-error source))
+    "[" "Unexpected EOF. [at line 1, column 2]"
+    "[}" "Unmatched delimiter: } [at line 1, column 2]"
+    "#" "Unexpected EOF. [at line 1, column 2]"
+    ":" "unexpected EOF while reading keyword. [at line 1, column 2]"
+    "\"" "Unexpected EOF while reading string. [at line 1, column 2]"
+    "#?" ":reader-macro node expects 1 value. [at line 1, column 3]"
+    "#:" "Unexpected EOF. [at line 1, column 3]"))
```

Comment:
It took me too long to find what source code would produce this error message :)

Reply:
https://github.com/borkdude/clj-kondo/blob/97b3ddcdbc63e70faa526641ba066ae0db174725/parser/clj_kondo/impl/rewrite_clj/reader.clj#L76